lib/src/test/java/uk/gov/di/ipv/cri/kbv/api/security/SoapTokenRetrieverTest.java

```diff
@@ -0,0 +1,82 @@
+package uk.gov.di.ipv.cri.kbv.api.security;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.time.Instant;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class SoapTokenRetrieverTest {
+    private static final String MOCKED_VALID_TOKEN_VALUE =
+            String.format(
+                    "{\"exp\": \"%s\"}",
+                    Instant.now().getEpochSecond() + TimeUnit.DAYS.toSeconds(1));
+
+    private static final String MOCKED_EXPIRED_TOKEN_VALUE =
+            String.format("{\"exp\": \"%s\"}", 0);
+
+    @Mock private SoapToken soapTokenMock;
+    @InjectMocks private SoapTokenRetriever soapTokenRetriever;
+
+    @Test
+    void shouldReturnNullWhenTokenIsNull() {
+        when(soapTokenMock.getToken()).thenReturn(null);
+
+        String token = soapTokenRetriever.getSoapToken();
+        assertNull(token);
+    }
+
+    @Test
+    void shouldReturnEmptyStringWhenTokenIsEmpty() {
+        when(soapTokenMock.getToken()).thenReturn("");
+        String token = soapTokenRetriever.getSoapToken();
+        assertEquals("", token);
```

Comment:
Please use spacing, to imply `arrange - act - assert` this and the other test as well